src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,27 +2,34 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Arrays;
-import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class ProductBasket {
-    private List<Product> basket = new LinkedList<>();
+    private Map<String, LinkedList<Product>> basket = new TreeMap<>();
```

Comment:
ссылки должны быть всегда интерфейсные, чтобы соблюдать полиморфизм
```suggestion
    private Map<String, List<Product>> basket = new TreeMap<>();
```